app/Jobs/FetchExchangeDetails.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Jobs;
+
+use App\Contracts\MarketDataProvider;
+use App\Models\Exchange;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\Middleware\RateLimited;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Arr;
+
+final class FetchExchangeDetails implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    /**
+     * The maximum attempts for this job.
+     *
+     * @var int
+     */
+    public $tries = 3;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(public Exchange $exchange)
+    {
+        //
+    }
+
+    /**
+     * Get the middleware the job should pass through.
+     *
+     * @return array<int, object>
+     */
+    public function middleware(): array
+    {
+        return [new RateLimited('coingecko_api_rate')];
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $result = app(MarketDataProvider::class)->exchangeDetails($this->exchange);
+
+        // No information for ark in any usd compatible exchange
+        if ($result === null) {
+            $this->exchange->price  = null;
+            $this->exchange->volume = null;
+        } else {
+            $this->exchange->price  = Arr::get($result, 'price');
+            $this->exchange->volume = Arr::get($result, 'volume');
+        }
```

Comment:
Referring to this as the "forever" cache - can change it to the below so it only gets updated when we have a value
```suggestion
        if ($result !== null) {
            $this->exchange->price  = Arr::get($result, 'price');
            $this->exchange->volume = Arr::get($result, 'volume');
        }
```